source/sigv4.c

```diff
@@ -1972,53 +2034,50 @@ static void generateCredentialScope( const SigV4Parameters_t * pSigV4Params,
                                       pCanonicalRequest->pQueryLoc[ paramsIndex ].key.dataLen,
                                       pBufLoc,
                                       &encodedLen,
-                                      true,
-                                      false );
+                                      true /* Encode slash (/) */,
+                                      false /* Do not encode '='. */ );
 
             if( returnStatus == SigV4Success )
             {
                 pBufLoc += encodedLen;
                 remainingLen -= encodedLen;
 
-                /* Encode parameter value if non-empty. Query parameters can have empty values. */
-                if( pCanonicalRequest->pQueryLoc[ paramsIndex ].value.dataLen > 0U )
-                {
-                    assert( pCanonicalRequest->pQueryLoc[ paramsIndex ].value.pData != NULL );
-                    returnStatus = writeValueInCanonicalizedQueryString( pBufLoc,
-                                                                         remainingLen,
-                                                                         pCanonicalRequest->pQueryLoc[ paramsIndex ].value.pData,
-                                                                         pCanonicalRequest->pQueryLoc[ paramsIndex ].value.dataLen,
-                                                                         &encodedLen );
-                    pBufLoc += encodedLen;
-                    remainingLen -= encodedLen;
-                }
+                assert( pCanonicalRequest->pQueryLoc[ paramsIndex ].value.pData != NULL );
```

Comment:
@aggarw13 Since, as you mentioned above that, `writeValueInCanonicalizedQueryString` can work with empty values, why are we asserting this?

Doesn't it essentially force all parameters to have non-empty value?